eos/maths/gegenbauer-polynomial.cc

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023 Viktor Kuschke
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <eos/maths/power-of.hh>
+#include <eos/maths/gegenbauer-polynomial.hh>
+
+#include <cmath>
+#include <vector>
+
+namespace eos
+{
+    std::vector<double> GegenbauerPolynomial::calculate_coefficients() const
+    {
+        const double epsilon = 1e-8;
+        double entry;
+        std::vector<double> result (half_order + 1);
+
+        for (unsigned int i = 0 ; i <= half_order ; i++)
+        {
+            entry = pow(2.0, 2.0 * i + 1.0 * r) * pow(-1.0, 1.0 * (half_order - i)) *
+                tgamma(alpha + 1.0 * (half_order + i + r)) / (tgamma(1.0 * (half_order - i + 1)) * tgamma(2.0 * i + 1.0 * (r + 1))); // denominator: (half_order - i)! * (2i + r)!;
+
+            result.at(i) = entry;
+        }
+
+        if (std::abs(alpha) <= epsilon)
+            return result;
+        else
+            for (unsigned int i = 0 ; i <= half_order ; i++)
+            {
+                result[i] = result[i] / tgamma(alpha);
```

Comment:
You are making ``N = half_order + 1`` calls to ``tgamma`` here, although you need to evaluate it only once. How about the following.

```C++
double gamma_alpha = (alpha < epsilon) ? 1.0 : tgamma(alpha);
for (unsigned int i = 0 ; ...)
{
    result[i] /= gamma_alpha;
}
return result;
```